server/controllers/user.controller.js

```diff
@@ -144,6 +148,44 @@ class UserController {
       Response(res, 400, errorMessage);
     }
   }
+
+  /**
+     * @description updates the role of a user
+     * @param {object} req
+     * @param {object} res
+     * @returns {object} User Profile Object
+     */
+  static async updateUserRole(req, res) {
+    const userId = req.params.id;
+    const { newrole } = req.body;
+    const foundUser = await User.findByPk(userId);
+
+    if (!foundUser) return Response(res, 404, 'This user was not found.');
+
+    const newRoleId = await Role.findOne({ where: { type: newrole } });
```

Comment:
@TheDrizzyWay What happens when the role is not found in the Role table?

Reply:
More so, it makes more sense to search for the role by the id or uuid.

Reply:
@paradisekelechi there's a middleware that throws an error, when the role entered does not exist. The request will not reach the database. For the part of searching via uuid, do i still use req.body for that or do i use req.params instead?

Reply:
@paradisekelechi changes implemented and screenshot updated. please review